File: COOLSOUND-FRONTEND/coolsound/src/app/services/hilo.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Hilo } from '../model/hilo';

@Injectable({
  providedIn: 'root'
})
export class HiloService {

  private baseUrl = '/hilo/';

  constructor(private http: HttpClient) { }

  /** Método para sacar los Hilos de BD */
  hilos(): Observable<any> {
    return this.http.get<Hilo>(`${this.baseUrl}` + 'getHilos/');
  }

  /** Método para eliminar Hilos de BD */
  eliminarHilo(idHilo: number): Observable<any> {
    return this.http.get(`${this.baseUrl}` + 'eliminarHilo/' + idHilo);
  }

  /** Método para añadir Hilos a BD */
  nuevoHilo(nombreHilo: string, idUsuario: number): Observable<any> {
    return this.http.get(`${this.baseUrl}` + 'nuevoHilo/' + nombreHilo + '/' + idUsuario);
  }
}

File: COOLSOUND-BACKEND/coolsound/src/main/java/com/java/coolsound/service/HiloService.java
package com.java.coolsound.service;

import java.util.List;

import com.java.coolsound.model.Comentario;
import com.java.coolsound.model.Hilo;

public interface HiloService {

	/**
	 * Gets the all hilos.
	 *
	 * @return the all hilos
	 */
	List<Hilo> getAllHilos();
	
	/**
	 * Eliminar Hilo.
	 *
	 * @return eliminar hilo
	 */
	int eliminarHilo(int idHilo);
	
	/**
	 * Añadir Hilo.
	 *
	 * @return añadir hilo
	 */
	int nuevoHilo(String nombreHilo, int idUsuario);
	
	/**
	 * Gets a hilo.
	 *
	 * @return the hilo
	 */
	List<Comentario> getHilos(int idHilo);
}
